### PR TITLE
feat(linter): eslint-plugin-jsx-a11y img-redundant-alt (correctness)

### PR DESCRIPTION
**Pull request by Shinyaigeek**
Hi team 👋 I really love this project, so I tried to contribute this to help even a little

## Summary

partof: #1141

I re-implemented img-redundant-alt rule for jsx_a11y in Rust same as the original JS one, and moved also the test related to the rule to here.

originals:
- doc: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/docs/rules/img-redundant-alt.md 
- code: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/src/rules/img-redundant-alt.js
- test: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/main/__tests__/src/rules/img-redundant-alt-test.js

### CODE REVIEW

File: crates/oxc_linter/src/rules/jsx_a11y/img_redundant_alt.rs [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        (r"<img alt />", None),
+        (r"<img alt={imageAlt} />", None),
+        (r"<img alt={imageAlt.name} />", None),
+        (r"<img alt={imageAlt?.name} />", None),
+        (r"<img alt='Doing cool things' aria-hidden={foo?.bar}/>", None),
+        (r"<img alt='Photography' />;", None),
+        (r"<img alt='ImageMagick' />;", None),
+        (r"<Image alt='Photo of a friend' />", None),
+        // TODO we need components_settings to test this
+        // (r"<Image alt='Foo' />", settings: Some(components_settings))
```
Comment (Shinyaigeek) [inline crates/oxc_linter/src/rules/jsx_a11y/img_redundant_alt.rs:211]: this is same to [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/1adec3517fc2c9797212ca4d38858deed917e7be/__tests__/src/rules/img-redundant-alt-test.js#L26-L32).

I could not find any way to use componentsSettings in linter testing, but I will really appreciate if the way actually exists and you teach me

File: crates/oxc_linter/src/snapshots/img_redundant_alt.snap [hunk 2]
```diff
@@ -0,0 +1,189 @@
+---
+source: crates/oxc_linter/src/tester.rs
+expression: img_redundant_alt
+---
+  ⚠ eslint-plugin-jsx-a11y(img-redundant-alt): Redundant alt attribute. Screen-readers already announce `img` tags as an image. You don’t need to use the words `image`, `photo,` or `picture` (or any
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/img_redundant_alt.snap:5]: Shall we move the "Screen-readers ..." sentence to help message?

File: crates/oxc_linter/src/snapshots/img_redundant_alt.snap [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  │ specified custom words) in the alt prop.
+   ╭─[img_redundant_alt.tsx:1:1]
+ 1 │ <img alt='Photo of friend.' />;
+   · ──────────────────────────────
```
Comment (Boshen) [inline crates/oxc_linter/src/snapshots/img_redundant_alt.snap:9]: The `alt` attribute is the only interesting bit, shall we point the span to `alt` only? Or even better `alt` and `Photo`.

**Review by Boshen — CHANGES_REQUESTED** [review]

File: crates/oxc_linter/src/snapshots/img_redundant_alt.snap (hunk 3 above)
Comment (Shinyaigeek) [reply crates/oxc_linter/src/snapshots/img_redundant_alt.snap]: your point makes sense, thanks ! [b1e5991](https://github.com/oxc-project/oxc/pull/1571/commits/b1e5991499e78829ad1d82c0adf35751ad32646e) 👍

File: .typos.toml [hunk 4]
```diff
@@ -10,6 +10,7 @@ extend-exclude = [
   "tasks/prettier_conformance/prettier",
   "crates/oxc_parser/src/lexer/mod.rs",
   "crates/oxc_linter/fixtures",
+  "crates/oxc_linter/src/rules/jsx_a11y/img_redundant_alt.rs",
```
Comment (Shinyaigeek) [inline .typos.toml:13]: https://github.com/oxc-project/oxc/pull/1571/files#diff-d06ab4ebdb38671afdf28c5d9b48cfc2e5a80255b1d73eaa7324f7df9338b576R231

some test case in img_redundant_alt rule has intentional typo in file, and I could not find any way to disable typo check within per one line, so I ignore `crates/oxc_linter/src/rules/jsx_a11y/img_redundant_alt.rs`...

**Review by Boshen — APPROVED** [review]

**Pull request merged**